node/resolvers/search.ts

```diff
@@ -0,0 +1,214 @@
+import {
+  Client,
+  Status,
+  PlaceAutocompleteType,
+  Language,
+  AddressType,
+  Place,
+  AddressGeometry,
+} from '@googlemaps/google-maps-services-js'
+import { pipe } from 'ramda'
+
+import countryRules from '../countries/rules'
+import { toAlpha3 } from '../countries/ISO'
+import { ISOAlpha2 } from '../countries/types'
+
+function revertRuleIndex(geolocationRules: GeolocationRules) {
+  return Object.entries(geolocationRules).reduce<{
+    [key: string]: GeolocationComponents
+  }>((acc, [propName, rule]) => {
+    for (let i = 0; i < rule!.types.length; i++) {
+      const type = rule!.types[i]
+      acc[type] = propName as GeolocationComponents
+    }
+    return acc
+  }, {})
+}
+
+function getCheckoutFieldName(
+  types: string[],
+  indexedRules: { [key: string]: GeolocationComponents }
+) {
+  const mappedType = types.find(type => indexedRules[type])
+  return mappedType ? indexedRules[mappedType] : null
+}
+
+// eslint-disable-next-line max-params
+function setAddressFieldValue(
+  address: Address,
+  fieldName: GeolocationComponents,
+  geolocationRules: GeolocationRules,
+  addressComponent: google.maps.GeocoderAddressComponent
+) {
+  const geolocationField = geolocationRules[fieldName]
+  address[fieldName] = addressComponent[geolocationField!.valueIn]
+  return address
+}
+
+function getCountry(place: Place) {
+  const countryComponent = place?.address_components?.find(
+    component => component.types.indexOf(AddressType.country) !== -1
+  )
+
+  return countryComponent
+    ? toAlpha3(countryComponent.short_name as ISOAlpha2)
+    : null
+}
+
+export const queries = {
+  getAddressByExternalId: async (
+    _: unknown,
+    args: { id: string },
+    ctx: Context
+  ): Promise<Address> => {
+    const { clients } = ctx
+    const { id } = args
+    const { apiKey } = await clients.apps.getAppSettings(
+      process.env.VTEX_APP_ID
+    )
+
+    const client = new Client({})
+
+    const response = await client.placeDetails({
+      params: {
+        place_id: id,
+        language: Language.pt_BR,
+        key: apiKey,
+      },
+      timeout: 1000,
+    })
+
+    if (response.statusText !== Status.OK) {
+      console.error(response.statusText)
+    }
+
+    const { result: place } = response.data
+
+    const baseAddress: Address = {
+      addressId: '1',
+      addressType: 'residential',
+      city: null,
+      complement: null,
+      country: null,
+      geoCoordinates: [],
+      neighborhood: null,
+      number: null,
+      postalCode: null,
+      receiverName: null,
+      reference: null,
+      state: null,
+      street: null,
+    }
+
+    const country = getCountry(place)
+
+    const rules = countryRules[country!]
+
+    if (!rules) {
+      console.warn(
```

Comment:
Use `ctx.logger.warn` so we can get this warnings logged to Splunk

Reply:
hmm, all there is in the `ctx` is `metricsLogger`, which have a `add` and `getSummaries` methods. Do you know if the equivalent of what you suggested would be `ctx.metricsLogger.add(`?

Reply:
oh, sorry! it is `ctx.vtex.logger` if I'm not mistaken